tests/scripts/all.sh

```diff
@@ -271,6 +275,7 @@ General options:
      --random-seed      Use a random seed value for randomized tests (default).
   -r|--release-test     Run this script in release mode. This fixes the seed value to ${RELEASE_SEED}.
   -s|--seed             Integer seed value to use for this test run.
+  --new-bignum-interface    Use the new bignum interface instead of the old one.
```

Comment:
I don't understand the testing approach here. Why would we test the new bignum by running the whole CI, and not by adding a few components (presumably a 64-bit one and a 32-bit one, both in the full config, what else would we need)?

Reply:
The added components are used as a regression test and this new option is used to set up a nightly job that does a full CI run on development with the new ECP module and only relevant for the bignum development.

Reply:
As discussed in today's call, I don't understand why we would use this approach.

It's a separate nightly job so if we break something in a pull request, we won't know it until the code has been merged. It's extra work to run and monitor this job, and the person monitoring the job won't necessarily know what's going on with the changes.

We don't expect changes to the new bignum itself, but it can be affected by global changes, e.g. to platform functions (like zeroization), to build scripts, to the test framework...

Why would we run the whole CI? Why not just run the unit tests in the full configuration — once on 64-bit and once on 32-bit? (Also little vs big endian once we have that.) I don't understand why we'd want test the full matrix of configurations against that particular option. We don't do that for everest and p256-m, why would we do it for this ECC implementation?

Reply:
As I wrote in https://github.com/Mbed-TLS/mbedtls-test/issues/103#issuecomment-1648163435, I am withdrawing this objection, given that my main concern (that we wouldn't test the new bignum in the regular CI) was based on a misconception. I still think this is a waste of effort, but it's not harmful if done right.